api/lib/read-graphs/source/isReferenced.ts

```diff
@@ -0,0 +1,22 @@
+import { ask } from '../../sparql'
+import { getClient } from '../sparqlClient'
+import { api, dataCube } from '../../namespaces'
+
+export function isReferenced (sourceId: string) {
+  return ask()
+    .where(`
+        ?s ?uses <${sourceId}>
+
+        MINUS {
+            ?s a dataCube:Project
+        }
+        MINUS {
+            <${sourceId}> api:columns ?s
+        }
+    `)
```

Comment:
Here's a SPARQL to check that the source is referenced.

It excludes reference by the parent project and its columns collection resource. Any other occurrence will prevent the delete